### PR TITLE
Add missing cast to fix build on i386

### PR DESCRIPTION
**Pull request by kontura**
From: https://download.copr.fedorainfracloud.org/results/rpmsoftwaremanagement/dnf5-unstable/fedora-37-i386/06330258-dnf5/builder-live.log.gz
```
/builddir/build/BUILD/dnf5-5.1.2-20230822004434.9.g54ab9891/include/libdnf5-cli/output/repo_info.hpp:67:86: error: conversion from 'long long int' to 'time_t' {aka 'long int'} may change value [-Werror=conversion]
   67 |         last_update = fmt::format("{:%F %X}", std::chrono::system_clock::from_time_t(cache_updated));
      |                                                                                      ^~~~~~~~~~~~~
```

It is failing in nightly: https://copr.fedorainfracloud.org/coprs/rpmsoftwaremanagement/dnf5-unstable/build/6350897/

### CODE REVIEW

File: include/libdnf5-cli/output/repo_info.hpp [hunk 1]
```diff
@@ -61,7 +61,7 @@ void RepoInfo::add_repo(Repo & repo) {
         add_line("Include packages", include_packages);
     }
 
-    auto cache_updated = repo.get_timestamp();
+    auto cache_updated = static_cast<time_t>(repo.get_timestamp());
```
[the remark this answers is not in the document]
Comment (ppisar) [reply include/libdnf5-cli/output/repo_info.hpp:64]: It indeed does not work. On i686 Fedora 40:

~~~~
<mock-chroot> sh-5.2# cat test.cpp 
#include <stdint.h>
#include <ctime>
#include <iostream>
int main(void) {
    int64_t value_64;
    std::time_t value_time_t;

    value_64 = INT64_MAX;
    value_time_t = static_cast<std::time_t>(value_64);

    std::cout << value_64 << "\n";
    std::cout << value_time_t << "\n";

    return 0;
}
<mock-chroot> sh-5.2# g++ -Werror=conversion test.cpp && ./a.out 
9223372036854775807
-1
~~~~

[the remark this answers is not in the document]
Comment (kontura) [reply include/libdnf5-cli/output/repo_info.hpp]: Looking where the timestamp comes from: 
Set in `read_metadata_cache()` -> which uses `mtime(..)` -> which uses `stat.3` accessing `struct timespec` member `tv_sec` which should have type `time_t`.

But in general you are right. I have added another commit with the logic.
Can you please take a look?

[the remark this answers is not in the document]
Comment (ppisar) [reply include/libdnf5-cli/output/repo_info.hpp]: It looks reasonable. But the original code treated cache_updated <= 0 specially. The thing is that both int64_t and time_t are signed types, hence std::in_range check will happily accept small negative values. Therefore your new implementation behaves differently.

I did not know whether the difference is important, so I looked into Repo class and found these cases for timestamp:

* -1 means undefined value (default value)
*  0 means expired (Repo::expire())
*  >= 0 means up-to-date (Repo::fresh())

I'm not sure whether the formatting code should respect them or not.

[the remark this answers is not in the document]
Comment (kontura) [reply include/libdnf5-cli/output/repo_info.hpp]: I see, yes for repos that don't have cache it prints `1969-12-31 23:59:59`. 
When running `dnf5 repo info --all` it also prints disabled repos that might not have cache. 

I have restored the check. Thanks!

[the remark this answers is not in the document]
Comment (ppisar) [reply include/libdnf5-cli/output/repo_info.hpp]: It looks good. I confirm it builds on i686.

**Review by ppisar — APPROVED** [review]

File: common/utils/string.hpp [hunk 2]
```diff
@@ -107,12 +110,13 @@ inline std::string tolower(const std::string & s) {
...
-    char buffer[20];
-    strftime(buffer, 20, "%F %T", ptm);
-    return std::string(buffer);
+template <typename T>
+inline std::string format_epoch(T epoch_num) {
+    if (std::in_range<time_t>(epoch_num)) {
+        const auto epoch = static_cast<time_t>(epoch_num);
+        return fmt::format("{:%F %X}", std::chrono::system_clock::from_time_t(epoch));
+    }
+    return "unrepresentable";
```
[the remark this answers is not in the document]
Comment (Conan-Kudo) [reply common/utils/string.hpp:119]: @kontura ping about this question?

[the remark this answers is not in the document]
Comment (kontura) [reply common/utils/string.hpp:119]: Sorry for the delay.

I like it. Here is a PR: https://github.com/rpm-software-management/dnf5/pull/891

**Pull request merged**